### DISCUSSION TITLE
How should helm chart be updated to support multiple access control types

### DISCUSSION THREAD

**Discussion by lozbrown** [trinodb/trino#24307] [General]
FYI @nineinchnick 

I've been looking at how we support ranger access control in the helm chart

At the moment the helm chart supports .Values.accessControl.type "configmap" or "properties"

the [ranger docs](https://trino.io/docs/current/security/apache-ranger-access-control.html) mention 

```
To combine Ranger access control with file-based or other access control systems, create the file etc/access-control.properties on the coordinator, with the following configuration that lists multiple access control configuration file paths:

> access-control.config-files=etc/trino/file-based.properties,etc/trino/apache-ranger.properties
Order the configuration files list in the desired order of the different systems for overall access control. Configure each access-control system in the specified files.
```
should we replace  .Values.accessControl.type with a list like  .Values.accessControl.types and try to somehow  build up the variable for 
access-control.config-files

and always map the configmap/properties data to file-based.properties even if its the only one?

^ this might be a breaking change for existing helm chart when upgrading

In general I'm trying to float ideas about the best way to go about this so interested in off the top of your head thoughts

**Comment by nineinchnick** [comment]
If you set `.Values.accessControl.type` to `properties` and set `.Values.accessControl.properties` to 
```properties
access-control.config-files=etc/trino/file-based.properties,etc/trino/apache-ranger.properties
```
you can then add these files in `.Values.coordinator.additionalConfigFiles` and they'll be mounted in `/etc/trino` on the coordinator. I don't think any changes are necessary in the chart, but we can add a new test to confirm that.
